include/pa_asio.h

```diff
@@ -70,6 +70,26 @@ extern "C"
 PaError PaAsio_GetAvailableBufferSizes( PaDeviceIndex device,
         long *minBufferSizeFrames, long *maxBufferSizeFrames, long *preferredBufferSizeFrames, long *granularity );
 
+/** Register callback to detect changes to buffer size.
+
+ This correponds to the ASIO kAsioBufferSizeChange message.  Note that an ASIO
+ driver may send a more general reset request when the user changes the buffer
+ size.
+
+ @param callback The function to call when ASIO buffer size changes.
+ It will receive the new buffer size as a parameter.
+*/
+void PaAsio_RegisterBufferSizeChangeCallback( void (*callback)(long newBufferSize) );
```

Comment:
You could declare a typedef for the callback function, similar to PaStreamCallback.

https://github.com/PortAudio/portaudio/blob/f7efb6b53cf134ec9165da7f6a3c0172371b820a/include/portaudio.h#L830